server/api/utils/clients/async_nuclio.py

```diff
@@ -0,0 +1,53 @@
+from mlrun.utils import logger
+import mlrun.utils
+from mlrun.config import config
+import mlrun.common.schemas
+import asyncio
+import urllib.parse
+import http
+import aiohttp
+
+NUCLIO_API_SESSIONS_ENDPOINT = "/api/sessions/"
+NUCLIO_API_GATEWAYS_ENDPOINT = "/api/api_gateways/"
+API_GATEWAY_NAMESPACE_HEADER = "X-Nuclio-Api-Gateway-Namespace"
+NUCLIO_PROJECT_NAME_HEADER = "X-Nuclio-Project-Name"
+
+
+class Client:
+    def __init__(self, auth_info: mlrun.common.schemas.AuthInfo):
+        self._auth = aiohttp.BasicAuth(auth_info.username, auth_info.session)
+        self._session = mlrun.utils.AsyncClientWithRetry(
+            raise_for_status=False, logger=logger
+        )
```

Comment:
as discussed earlier, this needs to be lazily initialized

from __init__.py
```py
self._session: typing.Optional[mlrun.utils.AsyncClientWithRetry] = None
```

then, when accessed, you can await to initializing the client upon need. look for
`async def _ensure_async_session(self):` piece of code